README.md

```diff
@@ -158,9 +159,47 @@ You can use the SharedPreferenceVault with SecretKey generated entirely from the
         SharedPreferenceVaultRegistry.getInstance().getVault(VAULT_ID).rekeyStorage(secretKey);
 
 #### Rekeying
-The vault can be rekeyed at any time. This will delete all values in the shared 
-preference file. This is completely irreversible.
+The vault can be rekeyed at any time. This will delete all values in the shared preference file. This is completely irreversible.
+
+#### API Changes to Commit and Apply
+Typical results of the SharedPreferenc commit and apply are slightly modified in Vault resulting of the extra encryption that is taking place. Because of this, commit and apply can both fail for reasons other than the SharedPreference write failing. The return type for commit is now more important. The recommendation is to **enable exceptions** when creating a SharedPreferenceVault.
+
+* `SharedPreferenceVaultFactory.getCompatAes256Vault()`
+* `SharedPreferenceVaultFactory.getAppKeyedCompatAes256Vault()`
+* `SharedPreferenceVaultFactory.getMemoryOnlyKeyAes256Vault()`
+
+These all have the ability to throw `GeneralSecurityException` or `UnsupportedEncodingException` with **exceptions enabled**. These exceptions can occur before the commit or apply is actually attempted and can result in a failure to write data to the SharedPreference file.
+
+1. If using `commit()` consider **enabling exceptions** to allow handling of these errors. Also, pay attention to the return type from `commit()` to make sure data was successfully written.
+2. If using `apply()`, there is a new `SharedPrefVaultWriteListener` interface that can be used to handle callback should any of the errors mentioned above. The `SharedPreferenceVault` interface now supports a new method to add this Listener to the Vault. `setSharedPrefVaultWriteListener(SharedPrefVaultWriteListener listener)` This method can be chained but must be called **before** an editor is created.
+    * The actual result of the write is still ignored when using the new `apply()` with a listener. The listener is only to show potential encryption issues that happen *before* the write.
+    * Setting no listener will behave like the standard `SharedPreference.apply()`. All error are ignored.
+    * The listener will only provide basic information on success and fail, it is still recommended that if more information is required, to **enable exceptions**.
```

Comment:
@lazysp333 nit: Suggested changes:
* The listener will only provide basic information on success and fail. It is recommended to **enable exceptions** if more information on the failure is required.